localstack/ext/java/src/main/java/cloud/localstack/Localstack.java

```diff
@@ -251,7 +272,10 @@ public void run() {
     protected void setupInfrastructure() {
         synchronized (INFRA_STARTED) {
             // make sure everything is installed locally
-            ensureInstallation();
+            ensureInstallation(true);
```

Comment:
Not sure if we want to pass `true` here? I think we may need a more robust approach to figure out whether we're executing in the dev environment.

For example, in `Localstack.ensureInstallation()` we could check if the git metadata in `../../../.git/config` points to the `localstack` repository, or something along those lines?

Reply:
As far as I can tell, we would want to run the setup `ensureInstallation(true);` from both the LocalstackTestRunner and the LocalstackTestExtension as any test will want to make sure they are using the latest java and python code.

Reply:
Sorry, should have made myself more clear. What I'm trying to say is - if we consider the "normal" usage of this library, **not** in our CI builds but in some Java project that the user wants to test using LocalStack, then it seems that the code in its current form would fail.

Consider the case where a user annotates their test with `@RunWith(LocalstackTestRunner.class)`, which then calls `setupInfrastructure()`, which then calls `ensureInstallation(true)`. Since a normal user would be using this library in their own project (i.e., the current working directory, `$PWD`, is not the LocalStack dev environment), we'd end up copying an arbitrary folder in the user's file system (`../../..`) in line 163. In other words, `localstackDir` in line 163 is not actually the LocalStack dir, but some arbitrary folder.

Does that make sense, or am I missing something here?

Reply:
That makes sense, I'll add it to the commit.